packages/mui-system/src/createStyled/createStyled.js

```diff
@@ -98,14 +95,15 @@ export default function createStyled(input = {}) {
     slotShouldForwardProp = shouldForwardProp,
   } = input;
 
-  const systemSx = (props) => {
-    return styleFunctionSx(attachTheme(props, themeId, defaultTheme));
-  };
-  systemSx.__mui_systemSx = true;
+  function styleAttachTheme(props) {
+    attachTheme(props, themeId, defaultTheme)
+  }
```

Comment:
I have removed the various `attachTheme` calls in favor of adding a style function that just attaches the theme where it goes. It relies on this running before anything else.